File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/TrgTolerance.java
package Auxiliary_Modules;

public class TrgTolerance extends TrGEV3{
    // this boolean gate will return true till depletion or reset()
    private int repeats = 0;
    private int maxrepeats; //2 recomended
    public TrgTolerance(int maxrepeats) {
        this.maxrepeats = maxrepeats;
    }

    public void setMaxrepeats(int maxrepeats) {
        this.maxrepeats = maxrepeats;
        reset();
    }

    @Override
    public void reset() {
        // refill trigger
        repeats = maxrepeats;
    }

    @Override
    public Boolean trigger() {
        // will return true till depletion or reset()
        repeats--;
        return repeats > 0;
    }
    public void disable(){
        repeats = 0;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/TrgTolerance.kt
package auxiliary_modules

class TrgTolerance( //2 recomended
    private var maxrepeats: Int
) : TrGEV3() {
    // this boolean gate will return true till depletion or reset()
    private var repeats = 0

    fun setMaxrepeats(maxrepeats: Int) {
        this.maxrepeats = maxrepeats
        reset()
    }

    override fun reset() {
        // refill trigger
        repeats = maxrepeats
    }

    override fun trigger(): Boolean {
        // will return true till depletion or reset()
        repeats--
        return repeats > 0
    }

    fun disable() {
        repeats = 0
    }
}